pymc/logprob/censoring.py

```diff
@@ -237,3 +241,147 @@ def round_logprob(op, values, base_rv, **kwargs):
     from pymc.math import logdiffexp
 
     return logdiffexp(logcdf_upper, logcdf_lower)
+
+
+class MeasurableSwitchEncoding(MeasurableElemwise):
+    """A placeholder used to specify the log-likelihood for a encoded RV sub-graph."""
+
+    valid_scalar_types = (Switch,)
+
+
+measurable_switch_encoding = MeasurableSwitchEncoding(scalar_switch)
+
+
+@node_rewriter(tracks=[switch])
+def find_measurable_switch_encoding(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[MeasurableSwitchEncoding]]:
+    rv_map_feature: Optional[PreserveRVMappings] = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    valued_rvs = rv_map_feature.rv_values.keys()
+
+    switch_condn, *components = node.inputs
+
+    # broadcasting of switch condition is not supported
+    if switch_condn.type.broadcastable != node.outputs[0].type.broadcastable:
+        return None
+
+    if rv_map_feature.request_measurable([switch_condn]) != [switch_condn]:
+        return None
+    # this automatically checks the measurability of the switch condition and converts switch to MeasurableSwitch
+
+    measurable_comp_list = [
+        idx
+        for idx, component in enumerate(components)
+        if check_potential_measurability([component], valued_rvs)
+    ]
+
+    # Maximum one branch allowed to be measurable
+    if len(measurable_comp_list) > 1:
+        return None
+
+    # If at least one of the branches is measurable
```

Comment:
Any chance that there can be no measurable branches following the previous checks? Then `len(measurable_comp_list) == 1` necessarily?